### PR TITLE
Fixing the issue of provisioner being override by torpedo params even it is  passed by ScheduleOptions from test and Adding support for provisioner volume snapshotclass mapping

### PR DESCRIPTION
**Pull request by sn-px**
**What this PR does / why we need it**:
Fixed the issue of provisioner getting override even though it is passed from ScheduleOptions
Adding common function for passing provisioner volumesnapshotclass mapping

Ignore reviewing the dummy test BasicBackupCreationDummyTest, which will be removed before merge 

**Which issue(s) this PR fixes** (optional)
Closes #https://portworx.atlassian.net/browse/PB-5533

**Special notes for your reviewer**:
https://jenkins.pwx.dev.purestorage.com/job/Users/job/santoshn/job/ocp-quick-byoc/26/execution/node/162/log/?consoleFull
https://aetos.pwx.purestorage.com/resultSet/testSetID/566432/testCaseID/3409909

Latest Log : https://jenkins.pwx.dev.purestorage.com/job/Users/job/santoshn/job/ocp-quick-byoc/87/execution/node/162/log/?consoleFull

Ran the basic test to ensure it is not breaking
https://jenkins.pwx.dev.purestorage.com/job/Users/job/santoshn/job/ocp-quick-byoc/23/consoleFull

Latest Log : https://aetos.pwx.purestorage.com/resultSet/testSetID/584295/testCaseID/3552299
After resolving merge conflict : https://jenkins.pwx.dev.purestorage.com/job/Users/job/santoshn/job/ocp-quick-byoc/90/console

### CODE REVIEW

**Review by mkoppal-px — CHANGES_REQUESTED** [review]
Please create a separate function instead of modifying the existing one

**Comment by sn-px** [comment]
> Please create a separate function instead of modifying the existing one

done

File: tests/backup/backup_sanity_test.go [hunk 1]
```diff
@@ -86,6 +86,288 @@ var _ = Describe("{UserGroupManagement}", func() {
 	})
 })
 
+// This Dummy testcase to test the provisioner support will be removed at the time of merge
+var _ = Describe("{BasicBackupCreationDummyTest}", func() {
+
+	var (
+		backupNames          []string
```
Comment (lsrinivas-pure) [inline tests/backup/backup_sanity_test.go:93]: Please remove this before merge

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Review by mkoppal-px — CHANGES_REQUESTED** [review]
As discussed, please pass the volume snapshot map in the backup Request itself and keep the function signature of `AdditionalBackupRequestParams` unchanged

**Comment by sn-px** [comment]
> As discussed, please pass the volume snapshot map in the backup Request itself and keep the function signature of `AdditionalBackupRequestParams` unchanged

-> Done

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Pull request merged**